lms/djangoapps/discussion_api/api.py

```diff
@@ -415,7 +415,7 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
         raise Http404
     num_pages = (resp_total + page_size - 1) / page_size if resp_total else 1
 
-    results = [CommentSerializer(response, context=context).data for response in responses]
+    results = [CommentSerializer(response, remove_fields=["children"], context=context).data for response in responses]
```

Comment:
@wajeeha-khalid @nasthagiri I may be misunderstanding something here, as I'm still trying to wrap my head around the discussion api / comment service, but are we making a recursive call to the client (`"recursive": True`, L381 and then removing the recursed fields from the response with the remove_fields line?  Seems like the children comments are potentially a lot of data to be sending over the wire and then throwing away, aren't they?